Added Get Annotation Shell Script

wsmithee opened the pull request:
This shell script returns either all of the annotations for a particular tenant or tenant + domain id for a specified period of time, or it returns the details for a particular annotation based on its annotation id.

wasmithee committed:
Added Get Annotation Shell Script

This shell script returns either all of the annotations for a particular tenant or tenant + domain id for a specified period of time, or it returns the detials for a particular annotatin based on annotation id.

msolnit commented on get-annotation.sh:
```diff
@@ -0,0 +1,162 @@
+        DATE_END=$(date +%s)000
+      fi
+      if [ -n "$DOMAIN_ID" ]; then
+        URL="https://$HOST/concerto/mpulse/api/annotations/v1?date-start=$DATE_START&date-end=$DATE_END&domain=$DOMAIN_ID"
```
@wsmithee the convention I've been following for variable names is upper-case for command-line params, and lower-case for everything else.  To keep with this convention, please rename the `URL` variable to `url`.

msolnit commented on get-annotation.sh:
```diff
@@ -0,0 +1,162 @@
+
+echo $URL
+curl --fail --silent --show-error -H "X-Auth-Token: $token_raw" -X GET $URL | jq
+curl_result=$?
```
@wsmithee this won't work; instead of getting the exit code for `curl`, you'll get the exit code for `jq`.  You'll need to instead capture the `curl` output to a variable, then check the exit code, then echo it to `jq` if curl succeeded.

msolnit commented on get-annotation.sh:
```diff
@@ -0,0 +1,162 @@
+curl_result=$?
+
+if [ $curl_result -ne 0 ]; then
+  echo "Failed to delete annotation (curl exit code: $curl_result)."
```
@wsmithee I'm assuming "delete" is a typo.

wsmithee replied:
Indeed. It is a typo

msolnit commented on get-annotation.sh:
```diff
@@ -0,0 +1,162 @@
+token_quoted=$(echo $token_json | jq .token)
+token_raw=$(echo $token_quoted | sed 's/\"//g')
+
+echo $URL
```
@wsmithee is this left over from some testing?  I don't think the utility should print out the URL it's using.  Consider that it may be incorporated into larger scripts that want to do additional processing on the output JSON.

wsmithee replied:
Yes.

msolnit commented:
@wsmithee thanks for this contribution!  I added some inline comments.  You also need to provide an update to `README.md` that documents how to use the new script.

wasmithee committed:
Fixed issues identified by Matt

1. Renamed URL variable to url in order to keep with established naming convention

2. Fixed issue with curl execution by capturing the curl output to a variable, then checking the exit code, then echoing it to jq if curl succeeded

3. Fixed typo

4. Removed debugging echo statement

wsmithee commented:
@msolnit, I've made the suggested changes to the code.  Thanks for reviewing this for me!

msolnit commented on get-annotation.sh:
```diff
@@ -149,13 +149,14 @@ fi
+  echo "Failed to retrieve annotation(s) (curl exit code: $curl_result)."
   exit 1
+else
+  echo $response | jq
```
@wsmithee wrap `$response` in quotes here to avoid problems with whitespace and other Bashy stuff.

wsmithee replied:
Will do.

msolnit commented:
@wsmithee Thanks for adding the docs!  Can you expand on this part and add some examples?

> You can also add an end timestamp to limit the results to a particular window of time and you can also specify a domain id to restrict results to a single mPulse app.

The options you've implemented are pretty flexible, which is great, and I want to make sure that all of the features are documented.
